assets/js/components/charts/SuccessRateLine.jsx

```diff
@@ -53,33 +53,28 @@ export default class SuccessRateLine extends Component<Props> {
   renderD3(initial = false) {
     const { width, height, data } = this.state
 
-    let srData = data.filter((d) => d.label === "Success rate")
-    if (!srData[0].values.length) {
+    if (!data){
       return
     }
-    srData[0].values.unshift({ time: srData[0].values[0].time, value: 0 })
-    srData[0].values.push({ time: srData[0].values[srData[0].values.length - 1].time, value: 0 })
-
-    const flatten = Array.prototype.concat(...srData.map((d) => [...d.values, ...d.forecast]))
 
     let initialTime, lastTime
 
-    if (!flatten || flatten.length < 1) {
+    if (!data.values || data.values.length < 1) {
       initialTime = new Date()
       lastTime = d3.timeMonth.offset(initialTime, 1)
     } else {
       initialTime = d3.timeHour.offset(
-        d3.min(flatten, (d) => d.time),
+        d3.min(data.values, (d) => d.time),
         -1
       )
-      lastTime = d3.timeMonth.offset(initialTime, 1)
-      lastTime = d3.max([d3.max(flatten, (d) => d.time), lastTime])
+      const oneMonthFromStart = d3.timeMonth.offset(initialTime, 1)
+      lastTime = d3.max([d3.max(data.values, (d) => d.time), oneMonthFromStart])
```

Comment:
We could eventually add a few days of leeway for still-running surveys, to hint that it hasn't finished yet.

Reply:
it's common to have surveys that last longer than one month?

Reply:
I think most full-scale surveys are around one month long now, but we used to have larger ones (a bit over two months).

Reply:
follow up on #2338